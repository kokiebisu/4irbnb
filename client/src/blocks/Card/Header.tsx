import React from 'react';
import { H } from '../../elements/H';

interface Props {
  children: React.ReactNode;
}

export const Header: React.FC<Props> = ({ children }) => {
  return <H>{children}</H>;
};
