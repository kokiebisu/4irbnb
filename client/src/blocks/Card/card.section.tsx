import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// element
import { Box } from '../../elements/Box';

// color
import { colorpallete } from '../../styles/colorpallete';

type CardProps = {
  title?: string;
  description?: string;
  option?: string;
};

export default ({ title, description, option, ...props }: CardProps) => {
  return (
    <Box styles={extend.wrapper} {...props}>
      <Box styles={extend.inner}>
        <Box styles={extend.text}>
          <Box styles={extend.title}>{title}</Box>
          <Box styles={extend.description}>{description}</Box>
          <Box styles={extend.option}>{option}</Box>
        </Box>
      </Box>
    </Box>
  );
};

const background = theme('mode', {
  light: colorpallete.black,
  dark: colorpallete.white,
});

const primary = theme('mode', {
  light: colorpallete.white,
  dark: colorpallete.black,
});

const extend = {
  wrapper: css`
    width: 100%;
    background-color: ${background};
    border-radius: 16px;
  `,
  inner: css`
    padding: 32px;
  `,
  text: css`
    padding-top: 56px;
    width: 100%;
    max-width: 360px;
    color: ${primary};
  `,
  title: css`
    font-size: 22px;
    font-weight: 700;
    margin-bottom: 8px;
    line-height: 1.3;
  `,
  description: css`
    line-height: 1.2;
  `,
  option: css`
    font-weight: 700;
    margin-top: 12px;
  `,
};
