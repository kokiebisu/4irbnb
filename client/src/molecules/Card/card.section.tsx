import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// atoms
import { Box } from 'atoms';

// colors
import { colorpallete } from 'styles/colorpallete';

type CardProps = {
  title?: string;
  description?: string;
  option?: string;
};

export default ({ title, description, option, ...props }: CardProps) => {
  return (
    <Box styles={extend.wrapper} {...props}>
      <Box styles={extend.inner}>
        <Box styles={extend.content.wrapper}>
          <Box styles={extend.content.title}>{title}</Box>
          <Box styles={extend.content.description}>{description}</Box>
          <Box styles={extend.content.option}>{option}</Box>
        </Box>
      </Box>
    </Box>
  );
};

const background = theme('mode', {
  light: colorpallete.black,
  dark: colorpallete.white,
});

const primary = theme('mode', {
  light: colorpallete.white,
  dark: colorpallete.black,
});

const extend = {
  wrapper: css`
    width: 100%;
    background-color: ${background};
    border-radius: 16px;
  `,
  inner: css`
    padding: 32px;
  `,
  content: {
    wrapper: css`
      padding-top: 56px;
      width: 100%;
      max-width: 360px;
      color: ${primary};
    `,
    title: css`
      font-size: 22px;
      font-weight: 700;
      margin-bottom: 8px;
      line-height: 1.3;
    `,
    description: css`
      line-height: 1.2;
    `,
    option: css`
      font-weight: 700;
      margin-top: 12px;
    `,
  },
};
