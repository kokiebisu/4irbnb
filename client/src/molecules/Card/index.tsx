import React from 'react';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';
import { darken } from 'polished';
import { Box, Text, Img } from 'atoms';
import { colors } from 'styles';

type CardProps = {
  image: string;
  title?: string;
  description?: string;
  subtitle?: string;
  size?: string;
};

const background = theme('mode', {
  light: colors.white,
  dark: darken(0.8, colors.gray),
});

const primary = theme('mode', {
  light: darken(0.8, colors.gray),
  dark: darken(0.4, colors.white),
});

const secondary = theme('mode', {
  light: colors.gray,
  dark: darken(0.4, colors.white),
});

const shadow = theme('mode', {
  light: 'rgba(0, 0, 0, 0.15)',
  dark: 'rgba(0, 0, 0, 0.15)',
});

const Wrapper = styled(Box)`
  height: 100%;
  background-color: ${background};
  border-radius: 16px;
  box-shadow: ${shadow} 0px 2px 8px;
`;

const ImageWrapper = styled(Box)`
  object-fit: cover;

  & > img {
    border-top-left-radius: 16px;
    border-top-right-radius: 16px;
  }
`;

const Content = styled(Box)`
  &.large {
    width: 50%;
    text-align: left;
    padding: 0 18px 12px 18px;
  }
  &.normal {
    padding: 0 18px 12px 18px;
  }
`;

const extend = {
  title: css`
    font-weight: 700;
    color: ${primary};
    padding: 4px 0 2px 0;
  `,
  subtitle: css`
    padding: 14px 0;
    color: ${primary};
    font-weight: 500;
    line-height: 1.3;
    font-size: 15px;
  `,
  description: css`
    padding-top: 2px;
    font-size: 13px;
    font-weight: 300;
    color: ${secondary};
    line-height: 1.4;
  `,
};

export default ({
  image,
  title,
  subtitle,
  description,
  size,
  ...props
}: CardProps) => {
  return (
    <Wrapper {...props}>
      <ImageWrapper>
        <Img src={image} alt='card-image' />
      </ImageWrapper>
      <Content className={size === 'lg' ? 'large' : 'normal'}>
        {title && <Text styles={extend.title}>{title}</Text>}
        {subtitle && <Text styles={extend.subtitle}>{subtitle}</Text>}
        {description && <Text styles={extend.description}>{description}</Text>}
      </Content>
    </Wrapper>
  );
};
