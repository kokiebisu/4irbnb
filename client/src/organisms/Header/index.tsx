import React from 'react';

import { LandingHeader } from 'organisms/Header/header.landing';

export type HeaderProps = {
  type?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

export const Header: React.Props<HeaderProps> = ({ type, ...props }) => {
  if (!type) {
    return null;
  }
  const types: mapProps = {
    landing: <LandingHeader {...props} />,
  };
  return types[type];
};
