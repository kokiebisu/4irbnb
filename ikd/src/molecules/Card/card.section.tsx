import React from 'react';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';
import { Text } from 'atoms';
import { Box } from 'atoms';
import { colors } from 'styles';

type CardProps = {
  title?: string;
  description?: string;
  option?: string;
};

const background = theme('mode', {
  light: colors.black,
  dark: colors.white,
});

const primary = theme('mode', {
  light: colors.white,
  dark: colors.black,
});

const Wrapper = styled(Box)`
  width: 100%;
  background-color: ${background};
  border-radius: 16px;
`;

const Inner = styled(Box)`
  padding: 32px;
`;

const Content = styled(Box)`
  padding-top: 56px;
  width: 100%;
  max-width: 360px;
  color: ${primary};
`;

const Title = styled(Text)`
  font-size: 22px;
  font-weight: 700;
  margin-bottom: 8px;
  line-height: 1.3;
`;

const Description = styled(Text)`
  line-height: 1.2;
`;

const Option = styled(Text)`
  font-weight: 700;
  margin-top: 12px;
`;

export default ({ title, description, option, ...props }: CardProps) => {
  return (
    <Wrapper {...props}>
      <Inner>
        <Content>
          <Title>{title}</Title>
          <Description>{description}</Description>
          <Option>{option}</Option>
        </Content>
      </Inner>
    </Wrapper>
  );
};
