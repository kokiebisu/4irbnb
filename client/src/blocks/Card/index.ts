import styled from 'styled-components';

import { Header } from './Header';

const Card: any = styled.div`
  background: #ffffff;
  border-radius: 2px;
  margin: 5px 5px 10px;
  padding: 5px;
  position: relative;
  box-shadow: 2px 2px 4px 0px #cfd8dc;
`;

Card.Header = Header;

export default Card;
