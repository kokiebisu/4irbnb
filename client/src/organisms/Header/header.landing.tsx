import React from 'react';
import theme from 'styled-theming';
import { darken, lighten } from 'polished';
import styled, { css } from 'styled-components';
import { Text, Box, Icon } from 'atoms';
import { CustomButton } from 'molecules/Button';
import { colors, sizes } from 'styles';

const fill = theme('mode', {
  light: colors.white,
});

const Left = styled(Box)``;

const Middle = styled(Box)``;

const Right = styled(Box)``;

export default () => {
  return (
    <Box>
      <Left>
        <Icon name='NoColorLogo' fillColor={fill} />
      </Left>
      <Middle>
        <Text>header</Text>
      </Middle>
      <Right></Right>
    </Box>
  );
};
