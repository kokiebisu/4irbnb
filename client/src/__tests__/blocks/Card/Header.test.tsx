import React from 'react';
import { shallow } from '../../../enzyme';
import renderer from 'react-test-renderer';
import { Header } from '../../../blocks/Card/Header';

describe('Basic Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header>header</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
