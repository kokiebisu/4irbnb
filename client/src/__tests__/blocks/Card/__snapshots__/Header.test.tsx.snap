// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Header renders correctly 1`] = `
<h3
  className="sc-AxjAm gxygnu"
  style={Object {}}
>
  header
</h3>
`;
